browser/js/common/directives/graphs/add-graph.directive.js

```diff
@@ -0,0 +1,77 @@
+app.directive('addGraph', function($http) {
+    return {
+        restrict: 'E',
+        templateUrl: 'js/common/directives/graphs/add-graph.directive.html',
+        scope: {
+            data: "=",
+            info: "=",
+            columns: "="
+        },
+        link: function(scope, ele, attrs) {
+
+        	function validateNumber(col) {
+        		let numArray = [];
+        		for (let i=0; i < scope.data.length; i++) {
+        			if (i === 10) break;
```

Comment:
Could add this condition to the for loop above it, i.e. `i < scope.data.length && i < 10`.